code/Controllers/CMSMain.php

```diff
@@ -2017,14 +1995,14 @@ public function archive($data, $form)
         return $this->getResponseNegotiator()->respond($this->getRequest());
     }
 
-    public function publish($data, $form)
+    public function publish(array $data, Form $form): HTTPResponse
     {
         $data['publish'] = '1';
 
         return $this->save($data, $form);
     }
 
-    public function unpublish($data, $form)
+    public function unpublish(array $data, Form $form): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array